src/arns.lua

```diff
@@ -387,9 +387,14 @@ local function getTotalArNSNamesBetweenTimestampsUnsafe(startTimestamp, endTimes
 		if record.type == "permabuy" then
 			totalActiveNames = totalActiveNames + 1
 		elseif record.type == "lease" and record.endTimestamp then
+			-- record is active between the start and end timestamps
 			if record.startTimestamp <= startTimestamp and record.endTimestamp >= endTimestamp then
 				totalActiveNames = totalActiveNames + 1
-			elseif record.endTimestamp + constants.gracePeriodMs >= endTimestamp then
+			elseif
+				-- record is in the grace period between the start and end timestamps
```

Comment:
Grace period has to do with the CURRENT timestamp. I don't know if you can tell that something is in grace period given two arbitrary timestamps in the way that this function seems to.

Reply:
there is an issue with the current implementation e can only know of the record.endTimestamp > startTimestamp, so this needs to be fixed

Reply:
[12d396b](https://github.com/ar-io/ar-io-network-process/pull/352/commits/12d396b35436049058af2b96a18d4defac5ac0ef)

Reply:
tweaked implementation in [560dda9](https://github.com/ar-io/ar-io-network-process/pull/352/commits/560dda98224d770dfeaacec77ea1fc9ebedb5395)